rotkehlchen/db/accounting_rules.py

```diff
@@ -204,6 +205,28 @@ def add_linked_setting(
                 f'Link of rule for {rule_property} to setting {setting_name} already exists',
             ) from e
 
+    def get_valid_entry_combinations(
+            self,
+            entries: list[HistoryBaseEntry] | list[tuple[int, HistoryBaseEntry]],
+    ) -> set[tuple[HistoryEventType, HistoryEventSubType, Optional[str]]]:
+        if len(entries) == 0:
+            return set()
+        combinations = {
+            (entry[1].event_type, entry[1].event_subtype, getattr(entry[1], 'counterparty', None))
+            if isinstance(entry, tuple)
+            else (entry.event_type, entry.event_subtype, getattr(entry, 'counterparty', None))
+            for entry in entries
+        }
+        query = 'SELECT type, subtype, counterparty FROM accounting_rules WHERE'
+        query += ' OR '.join(f'(type="{x[0]}" AND subtype="{x[1]}" AND counterparty="{NO_ACCOUNTING_COUNTERPARTY if x[2] is None else x[2]}")' for x in combinations)  # noqa: E501
```

Comment:
I believe is better to use `?` and then populate a list of bindings to use cursor.execute(query, bindings)

Reply:
i would have to do
```python
query += ' OR '.join('(type=? AND subtype=? AND counterparty=?' for _ in bindings)
```
where I would have had to process bindings to a flat map of `(type, subtype, counterparty) for x in combinations`, then run an execute

Reply:
You can do `.join(['(type=? AND subtype=? AND counterparty=?'] * len(bindings))`